### ISSUE TITLE
Put back overlay system in 3.0

### ISSUE THREAD

**Issue by sapetnioc** [brainvisa/casa-distro#182]
I removed the management of writable singularity image from casa-distro 3. But, even if it is more limited that expected, it is still useful sometimes. I will put it back but I propose a new API that requires explicit options to use overlays.

To create a new overlay partition, one would use :
```
casa_distro create_overlay ~/casa_distro/my_overlay.img size=100G
```

To use an overlay partition, one would add `overlay` option to `casa_distro run`. For instance :
```
<env>/bin/bv overlay=~/casa_distro/my_overlay.img root=yes bash
```

We may have to change the option parsing in `bv` since we decided to make them Posix like. But now, `bv <command>` is a kind of shortcut to `casa_distro run` and the code above should quickly work.

**Comment by denisri** [comment]
I agree on making it an option, this will allow [re]using environments without the overlay. This can be discussed however:
- `+` I personally use overlays to install additional software (using apt). In some applications I use them, in others I don't, but I sometimes want to use several instances of the container at the same time, which is not allowed by singularity overlays. Disabling them allows to use them simultaneously (without the overlay)
- `+` it would allow sharing overlays across several environments
- `-` it could create inconsistencies: using overlays over mismatching/incompatible system images, or using an environment with and without software / libraries which could make software behave unexpectedly differently, or compile things in an incompatible way, generate failures very difficult to debug. I easily imagine a novice developer asking for help and saying "I have compiled brainvisa, then it crashes somewhere at runtime": how do we understand that part of the programs can be compiled with an overlay replacing some libraries, and another part without it ?
- `-` options are longer to type and more error-prone. Maybe we could write it in the environment JSON and use it by default when it's specified there, and have another option to disable it ?

**Comment by sapetnioc** [comment]
For me an overlay is highly connected to an image. Using it with another image can lead to a mess. It's too dangerous. Therefore I propose to take into account @denisri remarks with the followint API :
```
casa_distro create_overlay image=~/casa_distro/casa_dev-ubuntu-18.04.sif name=my_overlay size=100G
```
Would create an overlay file named `~/casa_distro/casa_dev-ubuntu-18.04.my_overlay.img` that can be used with :
```
bv overlay=my_overlay root=yes bash
```
We may want a default name for overlay but I do not find an elegant way to add an option that could be boolean (for the default name) and string (for a custom name). The best thing I can imagine is to let the name to be empty(`name=` at creation) and to use `overlay=`.

**Comment by denisri** [comment]
One possibility is to have two options, one bool (say "overlay") to use the default one, and another string (say "overlay_name" or "overlay_file") to use the a non-default one.
